packages/ses/src/commons.js

```diff
@@ -378,3 +378,32 @@ export const FERAL_STACK_GETTER = feralStackGetter;
  * @type {((newValue: any) => void) | undefined}
  */
 export const FERAL_STACK_SETTER = feralStackSetter;
+
+// Test for async generator function syntax support.
+let AsyncGeneratorNewFunctionInstance;
+try {
+  // Wrapping one in an new Function lets the `hermesc` binary file
+  // parse the Metro js bundle without SyntaxError, to generate the
+  // optimised Hermes bytecode bundle, when `gradlew` is called to
+  // assemble the release build APK for React Native prod Android apps.
+  // Delaying the error until runtime lets us customise lockdown behaviour.
+  AsyncGeneratorNewFunctionInstance = new FERAL_FUNCTION(
+    'return (async function* AsyncGeneratorFunctionInstance() {})',
+  )();
+} catch (e) {
+  // @ts-expect-error ts(2339) Property 'jsEngine' does not exist on type 'Error'. However it exists on Hermes.
+  if (Error.prototype.jsEngine === 'hermes' && e.name === 'SyntaxError') {
+    // Swallows Hermes error `async generators are unsupported` at runtime.
+    // @ts-expect-error ts(2554) Expected 0 arguments, but got 1. It refers to the Web API Window object, but on Hermes we expect 1 argument.
+    // eslint-disable-next-line no-undef
+    print('SES: Skipping async generators, unsupported on Hermes');
+    // Note: `console` is not a JS built-in, so Hermes engine throws:
+    // Uncaught ReferenceError: Property 'console' doesn't exist
+    // See: https://github.com/facebook/hermes/issues/675
+    // However React Native provides a `console` implementation:
+    // https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/Core/InitializeCore.js
+  } else {
+    throw e;
+  }
+}
+export const AsyncGeneratorFunctionInstance = AsyncGeneratorNewFunctionInstance;
```

Comment:
Would this work?

```suggestion

const getAsyncGeneratorFunctionInstance = () => {
  // Test for async generator function syntax support.
  try {
    // Wrapping one in an new Function lets the `hermesc` binary file
    // parse the Metro js bundle without SyntaxError, to generate the
    // optimised Hermes bytecode bundle, when `gradlew` is called to
    // assemble the release build APK for React Native prod Android apps.
    // Delaying the error until runtime lets us customise lockdown behaviour.
    return new FERAL_FUNCTION(
      'return (async function* AsyncGeneratorFunctionInstance() {})',
    )();
  } catch (e) {
    // @ts-expect-error ts(2339) Property 'jsEngine' does not exist on type 'Error'. However it exists on Hermes.
    if (Error.prototype.jsEngine === 'hermes' && e.name === 'SyntaxError') {
      // Swallows Hermes error `async generators are unsupported` at runtime.
      // @ts-expect-error ts(2554) Expected 0 arguments, but got 1. It refers to the Web API Window object, but on Hermes we expect 1 argument.
      // eslint-disable-next-line no-undef
      print('SES: Skipping async generators, unsupported on Hermes');
      // Note: `console` is not a JS built-in, so Hermes engine throws:
      // Uncaught ReferenceError: Property 'console' doesn't exist
      // See: https://github.com/facebook/hermes/issues/675
      // However React Native provides a `console` implementation:
      // https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/Core/InitializeCore.js
    } else {
      throw e;
    }
}
export const AsyncGeneratorFunctionInstance = getAsyncGeneratorFunctionInstance();
```

Reply:
i like it ^ updated w couple changes in https://github.com/endojs/endo/pull/2334/commits/a8e65934080b280d18b3e9cdce338883d59126c0 and https://github.com/endojs/endo/pull/2334/commits/6e1cb4994a85f5b2dc53e87fd4abda30f2c6beb2